junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/TestFactoryTestDescriptor.java

```diff
@@ -88,33 +84,20 @@ protected void invokeTestMethod(JupiterEngineExecutionContext context, DynamicTe
 					dynamicTest -> registerAndExecute(dynamicTest, index.incrementAndGet(), dynamicTestExecutor));
 			}
 			catch (ClassCastException ex) {
-				throw invalidReturnTypeException(testExtensionContext);
+				throw invalidReturnTypeException(ex);
 			}
 		});
 	}
 
 	@SuppressWarnings("unchecked")
 	private Stream<DynamicTest> toDynamicTestStream(TestExtensionContext testExtensionContext,
```

Comment:
It is still used by `executableInvoker.invoke` but it might be slightly better if it was moved inside the lambda's scope.